crates/cli/src/subcommands/generate/csharp.rs

```diff
@@ -430,69 +431,152 @@ fn autogen_csharp_access_funcs_for_struct(
             output,
             "public {csharp_field_name_pascal}UniqueIndex {csharp_field_name_pascal} = new();"
         );
-        writeln!(output);
     }
 
-    for idx in &schema.indexes {
-        match &idx.index_algorithm {
-            IndexAlgorithm::BTree(BTreeAlgorithm { columns }) => {
-                let col_pos = columns.head().unwrap().idx();
-                if constraints[&ColList::new(col_pos.into())].has_unique() {
-                    continue;
+    for index in btrees {
+        let fields: Vec<BTreeColumn<'_>> = index.columns
+            .iter()
+            .map(|col| {
+                let field = &product_type.elements[col.idx()];
+                let field_type = &field.algebraic_type;
+                match csharp_field_type(field_type) {
+                    None => todo!(),
+                    Some(x) => {
+                        let field_name = field.name.as_ref().expect("autogen'd tuples should have field names");
+                        let csharp_field_name_pascal = field_name.replace("r#", "").to_case(Case::Pascal);
+                        let range = match field_type {
+                            AlgebraicType::Product(pt) =>
+                                if pt.elements[0].name == Some("__address_bytes".into()) { ("Address.MinValue", "Address.MaxValue") }
```

Comment:
We should reuse the `is_address` and other helpers so that we don't hardcode those names in more places.